src/coil_read_write.cpp

```
@@ -0,0 +1,182 @@
+        response) {
+  static const uint8_t fc = MODBUS_FC_WRITE_COIL_CONTINUOUS;
+  //if (request->value == 1){
+  //  request->value = 0xFF00;}
+
+  uint8_t request_size =  (request->count + 7)/8; // count is number of coils ,data size in bytes
+  int loop_counter = 0;
+  std::vector<uint8_t> values_ ;
+  for (int i = 0; i < request_size; i++ )
+  {
+      for (int j = 0; j<8  ;j++){
+        values_[i] = (values_[i] << 1) | request->values[j];
```

cppcheck (2.10-2)

[error] Out of bounds access in expression 'values_[i]' because 'values_' is empty and 'i' may be non-zero.

(containerOutOfBounds)